### PR TITLE
Migrate QueryBuilder / ClassMetadataInfo

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
I baselined the following errors:

```
RROR: RedundantCondition - lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:602:13 - Operand of type Doctrine\ORM\Id\AbstractIdGenerator is always true (see https://psalm.dev/122)
        if ($parent->idGenerator) {


ERROR: RedundantCondition - lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:602:13 - Type Doctrine\ORM\Id\AbstractIdGenerator for $parent->idGenerator is never falsy (see https://psalm.dev/122)
        if ($parent->idGenerator) {


ERROR: InvalidNullableReturnType - lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1108:21 - The declared return type 'ReflectionClass<T>' for Doctrine\ORM\Mapping\ClassMetadataInfo::getReflectionClass is not nullable, but 'ReflectionClass<T>|null' contains null (see https://psalm.dev/144)
    public function getReflectionClass()


ERROR: RedundantCondition - lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1211:16 - array{columnDefinition?: string, columnName: string, declared?: class-string, declaredField?: string, enumType?: class-string<BackedEnum>, fieldName: string, generated?: string, id?: bool, inherited?: class-string, length?: int, notInsertable?: bool, notUpdatable?: bool, nullable?: bool, options?: array<string, mixed>, originalClass?: class-string, originalField?: string, precision?: int, quoted?: bool, scale?: int, type: string, unique?: string} can never contain false (see https://psalm.dev/122)
        return $mapping !== false && isset($mapping['unique']) && $mapping['unique'];


ERROR: RedundantCondition - lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1225:16 - array{columnDefinition?: string, columnName: string, declared?: class-string, declaredField?: string, enumType?: class-string<BackedEnum>, fieldName: string, generated?: string, id?: bool, inherited?: class-string, length?: int, notInsertable?: bool, notUpdatable?: bool, nullable?: bool, options?: array<string, mixed>, originalClass?: class-string, originalField?: string, precision?: int, quoted?: bool, scale?: int, type: string, unique?: string} can never contain false (see https://psalm.dev/122)
        return $mapping !== false && isset($mapping['nullable']) && $mapping['nullable'];


ERROR: TypeDoesNotContainType - lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1752:21 - Operand of type false is always false (see https://psalm.dev/056)
                if (! $this->table) {


ERROR: TypeDoesNotContainType - lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1752:23 - Type array{indexes?: array<array-key, mixed>, name: string, options?: array<string, mixed>, quoted?: bool, schema?: string, uniqueConstraints?: array<array-key, mixed>} for $this->table is never falsy (see https://psalm.dev/056)
                if (! $this->table) {
```

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 1]
```diff
@@ -1296,7 +1204,7 @@ public function isIdentifier(string $fieldName): bool
      *
      * @return bool TRUE if the field is unique, FALSE otherwise.
      */
-    public function isUniqueField($fieldName)
+    public function isUniqueField(string $fieldName): bool
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1198]: `@param` can be removed. `@return` too.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 2]
```diff
@@ -1310,7 +1218,7 @@ public function isUniqueField($fieldName)
      *
      * @return bool TRUE if the field is not null, FALSE otherwise.
      */
-    public function isNullable($fieldName)
+    public function isNullable(string $fieldName): bool
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1205]: `@param` can be removed. `@return` too.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 3]
```diff
@@ -1326,7 +1234,7 @@ public function isNullable($fieldName)
      *
      * @return string The column name.
      */
-    public function getColumnName($fieldName)
+    public function getColumnName(string $fieldName)
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: `@param` can be removed. `@return` too, after the return type was added.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 4]
```diff
@@ -1391,7 +1299,7 @@ public function getAssociationMappings()
      *
      * @return string The column alias.
      */
-    public function getFieldName($columnName)
+    public function getFieldName(string $columnName)
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: `@param` can be removed. `@return` too, after return type was added.

Maybe other methods are similar in this class.

**Review by SenseException — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:1198]: I hadn't reevaluated the comments but indeed, it's pretty tautological :sweat_smile:

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 5]
```diff
@@ -3280,12 +3017,10 @@ public function getAssociationMappedByTargetField(string $assocName): string
     }
 
     /**
-     * @param string|null $className
-     *
      * @return string|null null if the input value is null
      * @psalm-return class-string|null
      */
-    public function fullyQualifiedClassName($className)
+    public function fullyQualifiedClassName(string|null $className)
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]:
```suggestion
    public function fullyQualifiedClassName(string|null $className): string|null
```

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 6]
```diff
@@ -3298,12 +3033,8 @@ public function fullyQualifiedClassName($className)
...
     }
 
-    /**
-     * @param string $name
-     *
-     * @return mixed
-     */
-    public function getMetadataValue($name)
+    /** @return mixed */
+    public function getMetadataValue(string $name)
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]:
```suggestion
    public function getMetadataValue(string $name): mixed
```

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 7]
```diff
@@ -854,27 +806,18 @@ public function setIdentifierValues($entity, array $id)
...
     /**
      * Gets the specified field's value off the given entity.
      *
-     * @param object $entity
-     * @param string $field
-     *
      * @return mixed
      */
-    public function getFieldValue($entity, $field)
+    public function getFieldValue(object $entity, string $field)
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]:
```suggestion
    public function getFieldValue(object $entity, string $field): mixed
```

**Review by SenseException — CHANGES_REQUESTED** [review]

**Review by SenseException — APPROVED** [review]

**Pull request merged**